### PR TITLE
Add blade support

### LINKED ISSUE #8562: Laravel Blade support
Blade is a templating language for php by the laravel framework which also supports raw or inline php.

Here is an example for vim: https://github.com/alexandersix/vim-blade

### PR DESCRIPTION
**Pull request by lelgenio**
Fixes #8562

Add [`blade`](https://github.com/EmranMR/tree-sitter-blade) tree-sitter config.

It needs [`php_only`](https://github.com/tree-sitter/tree-sitter-php/tree/master/php_only), which is separate from the regular `php` grammar.

image

### CODE REVIEW

**Comment by lelgenio** [comment]
I can't figure out how to make `blade` be the default language for `.blade.php` files, instead of `php` :confused:

File: languages.toml
Comment (the-mikedavis) [inline languages.toml:898]: See https://github.com/helix-editor/helix/blob/81ae768a4ea20543bc16246a46552a69204fd7c1/languages.toml#L2945-L2948 as an example. When we detect the extension we split only the last `.<ext>` so you need to use a `suffix` configuration as a workaround

File: languages.toml
Comment (the-mikedavis) [inline languages.toml:891]: This language can be "hidden" since it's only used by injections:

```suggestion
name = "php-only"
scope = "source.php-only"
file-types = []
indent = { tab-width = 4, unit = "    " }
```

File: languages.toml
Comment (the-mikedavis) [inline languages.toml:894]:
```suggestion
name = "php-only"
```

File: runtime/queries/php_only/highlights.scm
Comment (the-mikedavis) [inline runtime/queries/php_only/highlights.scm:37]: `@property` should be replaced with `@variable.other.member` to match our scopes: https://docs.helix-editor.com/master/themes.html#syntax-highlighting

File: runtime/queries/php_only/highlights.scm
Comment (the-mikedavis) [inline runtime/queries/php_only/highlights.scm:70]:
```suggestion
(boolean) @constant.builtin.boolean
(null) @constant.builtin
(integer) @constant.numeric.integer
(float) @constant.numeric.float
```

File: languages.toml
Comment (lelgenio) [reply languages.toml:898]: This is not enough, regular `php` still takes precedence because suffixes are checked last https://docs.helix-editor.com/languages.html#file-type-detection-and-the-file-types-key.

File: languages.toml
Comment (the-mikedavis) [inline languages.toml:884]:
```suggestion
name = "php-only"
```

Also the `runtime/queries/php_only` dir should be renamed as `runtime/queries/php-only`. That should fix the injected php content

File: languages.toml
Comment (the-mikedavis) [reply languages.toml]: Hmm, we might need to change `helix-core/src/syntax.rs` to check suffixes first since they're more specific. Though we also want to switch to globs and I believe that PR handles this correctly: https://github.com/helix-editor/helix/pull/8006. I'll give that PR another look when I get the chance

**Comment by Agilan989** [comment]
One more question before this I used android phone they asking anything to
connect to iPhone

Thank you
Best regards
Agilanberan

File: languages.toml
Comment (lelgenio) [reply languages.toml]: Yup, I was grepping left and right, thanks! :smile:

File: languages.toml
Comment (the-mikedavis) [reply languages.toml]: I'll add a label that this PR is waiting on another PR. I would hold off on rebasing until that PR is merged so if we need to make any changes you don't have to deal with the conflicts

**Comment by Agilan989** [comment]
Hi helix

According the way I see it all because of language I can understand y
because first off falls the hacker grap my and hacked my android take all
the information MetaMask ,trust wallet, my coins tron wallet so many’s
token from my android after that he needed my information and details he
started hacking my iPhone .. once started hacking I was able to find I
thought it’s safari upgrade software.. whenever I call or talking to
someone used call it’s was declined whenever I go thru to investment link
it’s was show that indicator app was sistem update and I was not available
to download after that all was satirically by URL AND LET ME.. wait .. now
I need help first of fall need to forsage platforms used trust wallet
stolen my money and coins all was there MetaMask all was blocked .. I know
my Binance account aso available balance $$ and ETH and Bitcoin and then
and was not available to use some delete it but from begging until now I
used same email to create also investment .. android I used still was with
me but no sim usually my relatives son used for seeing movie used home WiFi
net that’s time this all happened.. please help me out sir I need to get my
money back please.. today back oso used chrome browser for withdrawal
request used Coinbase still I can saw my balance and how much coin I have
.. but still manage to create 2FA service from google help but still hacker
can be able to stolen from my wallet used iPhone 11(s) —its I was see in
security check and login check .. please sir help me out thank you very
much for your service

Thank you
Best regards
Agilanberan

File: languages.toml
Comment (lelgenio) [reply languages.toml]: Now that #8006 is merged, I rebased on master and everything should be working fine! :smiley:

File: runtime/queries/php-only/injections.scm
Comment (the-mikedavis) [inline runtime/queries/php-only/injections.scm:2]:
```suggestion
  (#set! injection.language "comment"))
```

We don't currently have a `phpdoc` language. We can use `comment` which has some basic highlighting for `TODO`, `HACK`, etc.

File: runtime/queries/blade/injections.scm
Comment (the-mikedavis) [inline runtime/queries/blade/injections.scm:11]: `#has-ancestor?` / `#not-has-ancestor?` is a predicate that only exists in nvim as far as I can tell. For now we can remove that line and always inject `php`

**Review by the-mikedavis — APPROVED** [review]
Thanks!

File: languages.toml
Comment (lelgenio) [reply languages.toml:935]: Now that https://github.com/helix-editor/helix/pull/8006 was merged, the blade language is being detected correctly.

https://github.com/helix-editor/helix/pull/9513#discussion_r1505137606

File: runtime/queries/blade/injections.scm
Comment (lelgenio) [reply runtime/queries/blade/injections.scm:1]: Oh, you're right! `@task`s don't work correctly. I wonder if there's an equivalent predicated to `#has-ancestor` in helix :thinking: 

image


https://github.com/helix-editor/helix/pull/9513#discussion_r1505262833

Comment (the-mikedavis) [reply runtime/queries/blade/injections.scm:1]: Seems like `#has-ancestor?` is working towards recognizing arbitrary nesting https://github.com/tree-sitter/tree-sitter/discussions/981

It was introduced in https://github.com/neovim/neovim/pull/23606 and scans from the currently captured node potentially up to the root. I don't think that's particularly expensive but I was imagining that when arbitrary nesting was supported upstream by tree-sitter itself (probably a new syntax in the queries) it would track heritage internally and eliminate the cost of scanning.

I will write up an issue for this when I get a chance but let's not block this PR on it since the design is up for debate

Comment (EmranMR) [reply runtime/queries/blade/injections.scm:1]: @the-mikedavis Absolutely, by all means! The `bash` injection and envoy is for a _very_ specific and niche use case in `blade` and its existence or lack of would not do any harm :). In fact no parser out there, had ever had this feature or been able to parse `shell` correctly in the `blade` files due to their limitation/complexity. In all honesty, I was just trying to see how far I can push `tree-sitter` with this feature when I was writing the grammar for it haha (tree-sitter never fails to please!).
But yea, having that section removed or commented for now, before the merge, is very sensible. Just to avoid unintended behaviour, because in the `injection.scm` both `bash` and `php` were injected to the same node.

And I am totally with you on that one, I wished 'tree-sitter' had some form of standardisation for the predicates...

Comment (lelgenio) [reply runtime/queries/blade/injections.scm:1]: I just updated the PR to no longer include the bash injection, by the way.

**Review by the-mikedavis — APPROVED** [review]

**Review by pascalkuthe — APPROVED** [review]

**Pull request merged**
